include/OsmAndCore/NetworkRouteContext.h

```diff
@@ -0,0 +1,172 @@
+    {
+        HIKING,
+        BICYCLE,
+        MTB,
+        HORSE,
+        
+        Count
+    };
+
+    struct OSMAND_CORE_API NetworkRouteKey
+    {
+        NetworkRouteKey();
+        NetworkRouteKey(NetworkRouteKey & other);
+        NetworkRouteKey(const NetworkRouteKey & other);
+        NetworkRouteKey(int index);
+        virtual ~NetworkRouteKey();
+        RouteType type;
+        QSet<QString> tags;
+        inline bool operator == (const NetworkRouteKey & other) const
+        {
+            if (type != other.type)
+                return false;
+            if (tags.size() != other.tags.size())
+                return false;
+            bool tagsEqual = true;
+            for (auto & tag : tags)
+            {
+                if (!other.tags.contains(tag))
+                {
+                    tagsEqual = false;
+                    break;
+                }
+            }
+            return tagsEqual;
+        }
+        inline bool operator != (const NetworkRouteKey & other) const
+        {
+            if (type != other.type)
+                return false;
+            if (tags.size() != other.tags.size())
+                return false;
+            bool tagsNotEqual = false;
+            for (auto & tag : tags)
+            {
+                if (!other.tags.contains(tag))
+                {
+                    tagsNotEqual = true;
+                    break;
+                }
+            }
+            return tagsNotEqual;
+        }
+        inline bool operator > (const NetworkRouteKey & other) const
+        {
+            return (type > other.type || tags.size() > other.tags.size());
+        }
+        inline bool operator < (const NetworkRouteKey & other) const
+        {
+            return (type < other.type || tags.size() < other.tags.size());
+        }
+        inline operator int() const
+        {
+            const int prime = 31;
+            int result = 1;
+            int s = 0;
+            for (const QString & tag : constOf(tags))
+            {
+                s += qHash(tag);
+            }
+            result = prime * result + s;
+            result = prime * result + qHash(static_cast<int>(type));
+            return result;
+        }
+        /*inline operator const QString() const
+        {
+            QString hash = QString::number(static_cast<int>(type));
+            for (auto & s : tags)
+            {
+                hash += s;
+            }
+            return hash;
+        }*/
+    };
+
+    struct OSMAND_CORE_API NetworkRouteSegment
+    {
+        NetworkRouteSegment();
+        NetworkRouteSegment(NetworkRouteSegment & other);
+        NetworkRouteSegment(const NetworkRouteSegment & other);
+        NetworkRouteSegment(std::shared_ptr<const Road> road, NetworkRouteKey rKey, int start_, int end_);
+        virtual ~NetworkRouteSegment();
+        int start;
+        int end;
+        std::shared_ptr<const Road> robj;
+        NetworkRouteKey routeKey;
```

Comment:
on any const int start, int end, NetworkRouteKey routeKey, std::shared_ptr<const Road> robj
'OsmAnd::NetworkRouteSegment' cannot be assigned because its copy assignment operator is implicitly deleted

Problem with QVector, QList, QMap, QHash of append, insert methods